scripts/extensions/videoEditor/src/VideoTimeline/VideoTimeline.tsx

```diff
@@ -0,0 +1,253 @@
+        // get list thumbnail render in list thumbnails get from server
+        const video = this.props.video.current!;
+        let widthPic = 0;
+
+        if (isEmpty(thumbnails)) {
+            widthPic = video && (50 * video.clientWidth) / video.clientHeight;
+        } else {
+            widthPic = thumbnails[0].width;
+        }
+        const numberThumbnails = Math.floor(this.controlbar.current!?.offsetWidth / widthPic);
+        const duration = video?.duration ?? 0;
+        let thumbnailsRender: Array<IThumbnail> = [];
+        const per_delta_image =
+            thumbnails.length > 1 ? (thumbnails.length - 1) / numberThumbnails : duration / numberThumbnails;
+
+        for (let i = 0; i <= numberThumbnails; i++) {
+            let thumbnail: IThumbnail = {
+                url: '',
+                width: widthPic,
+                height: 50,
+            };
+
+            if (this.props.thumbnails && this.props.thumbnails.length > 0) {
+                thumbnail = this.props.thumbnails[Math.round(i * per_delta_image)];
+                thumbnail.url = thumbnail.url + `?t=${Math.random()}`;
+            }
+            thumbnailsRender.push(thumbnail);
+        }
+
+        this.setState({thumnailsRender: thumbnailsRender});
+    }
+    tick = () => {
+        // updates the current time state
+        if (this.props.video.current) {
+            let currentTime = this.props.video.current!.currentTime;
+
+            if (currentTime <= this.props.trim.end) {
+                this.setState({currentTime: currentTime});
+            } else if (this.props.trim.end > 0) {
+                this.props.video.current!.pause();
+            }
+        }
+    }
+    // drag and drop left and right bar.
+    handleDragStart(e: React.DragEvent<HTMLDivElement>) {
+        // hide drag ghost image
+        const img = new Image();
+
+        img.src = 'data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs=';
+        e.dataTransfer.setDragImage(img, 0, 0);
+        e.dataTransfer.setData('text/plain', '');
+    }
+
+    handleDragEnd = () => {
+        this.setVideoCurrentTime(this.PositionX);
+        this.props.onTrim(this.props.trim.start, this.props.trim.end, true);
+    }
+
+    getPositionInBar = (pX: number) => { // returns seconds
+        const controlbar = this.controlbar.current;
+
+        if (controlbar == null) {
+            throw new Error('control bar element is not present');
+        }
+
+        var position = (pX - controlbar.getBoundingClientRect().left) / controlbar.offsetWidth;
+
+        if (position > 1) {
+            return 1;
+        } else if (position < 0) {
+            return 0;
+        } else {
+            return Math.floor(position * 1000) / 1000;
+        }
+    }
+
+    handleTimelineClick = (e: React.MouseEvent) => {
+        let time = this.setVideoCurrentTime(e.clientX);
+
+        if (time < this.props.trim.start) {
+            this.props.onTrim(time, this.props.trim.end, false);
```

Comment:
We have update this. This logic use for UX, when user click outer of trim, it will set trim and current time at this position.
We call `onTrim` to update Trim in `videoEditor`, and `trimUpdate` for update state trim in `videoTimeline`.
![video-_2_](https://user-images.githubusercontent.com/45254285/71164284-53c89000-2281-11ea-9303-fa7ef9bb39e2.gif)

Reply:
got it, thanks